test/EmployeeVesting.js

```diff
@@ -0,0 +1,399 @@
+                assert.equal(userVesting[0].toString(), await web3.utils.toWei('50'))
+                assert.equal(userVesting[1].toString(), await web3.utils.toWei('0'))
+
+                time.increase(26 * 7 * 24 * 60 * 60)
+                await vesting.claim(0, { from: accounts[1] })
+                let claimed = await vesting.getVesting(accounts[1], 0)
+                assert.equal(claimed[0].toString(), await web3.utils.toWei('50'))
+                assert.equal(claimed[1].toString(), await web3.utils.toWei('50').toString())
+            })
+
+            it('With no cliff', async () => {
+                await tcr.transfer(vesting.address, web3.utils.toWei('50'))
+                await vesting.setVestingSchedule(accounts[1], web3.utils.toWei('50'), true, 0, 156, now);
+                let userVesting = await vesting.getVesting(accounts[1], 0)
+                assert.equal(userVesting[0].toString(), await web3.utils.toWei('50'))
+                assert.equal(userVesting[1].toString(), await web3.utils.toWei('0'))
+
+                time.increase(13 * 7 * 24 * 60 * 60)
+                await vesting.claim(0, { from: accounts[1] })
+                let claimed = await vesting.getVesting(accounts[1], 0)
+                assert.equal(claimed[0].toString(), await web3.utils.toWei('50'))
+                assert.equal(claimed[1].toString().slice(0, 6), await web3.utils.toWei('4.16666').toString().slice(0, 6))
+            })
+
+            it('Allows a user to claim against multiple schedules', async () => {
+                await tcr.transfer(vesting.address, web3.utils.toWei('150'))
+                //Setup multiple vesting schedules
+                await vesting.setVestingSchedule(accounts[1], web3.utils.toWei('50'), true, 26, 26, now);
+                await vesting.setVestingSchedule(accounts[1], web3.utils.toWei('100'), true, 26, 52, now);
+
+                time.increase(26 * 7 * 24 * 60 * 60)
+                let tokensBefore = await tcr.balanceOf(accounts[1])
+                await vesting.claim(0, { from: accounts[1] })
+                let claimed = await vesting.getVesting(accounts[1], 0)
+                assert.equal(claimed[0].toString(), await web3.utils.toWei('50'))
+                assert.equal(claimed[1].toString(), await web3.utils.toWei('50').toString())
+                await vesting.claim(1, { from: accounts[1] })
+                let claimed2 = await vesting.getVesting(accounts[1], 1)
+                assert.equal(claimed2[0].toString(), await web3.utils.toWei('100'))
+                assert.equal(claimed2[1].toString().slice(0, 6), await web3.utils.toWei('50').toString().slice(0, 6))
+                let tokensAfter = await tcr.balanceOf(accounts[1])
+                assert.equal(tokensAfter.sub(tokensBefore).toString().slice(0, 6), web3.utils.toWei('100').toString().slice(0, 6))
+            })
+        })
+    })
+    describe('cancel', () => {
```

Comment:
Check for `onlyOwner` is warranted here.

Reply:
@sporejack what do you mean by this comment? A check for onlyOwner and onlyDAO exists for this test?